src/Algebra/Module/Morphism/Linear/Properties.agda

```diff
@@ -0,0 +1,172 @@
+--
+-- Properties of linear maps.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra                   using (CommutativeRing)
+open import Algebra.Module            using (Module)
+open import Level                     using (Level; _⊔_)
+
+module Algebra.Module.Morphism.Linear.Properties
+  {r ℓr m ℓm : Level}
+  {ring      : CommutativeRing r ℓr}
+  (modA modB  : Module ring m ℓm)
+  where
+
+import Algebra.Module.Properties          as Properties
+import Function.Definitions
+import Relation.Binary.Reasoning.Setoid   as Reasoning
+import Algebra.Module.Morphism.Structures as MorphismStructures
+
+open import Agda.Builtin.Sigma
+open import Axiom.DoubleNegationElimination
+open import Data.List
+open import Data.Product
+  using (Σ; _,_; ∃; Σ-syntax; ∃-syntax; _×_)
+open import Relation.Nullary          using (¬_)
+open import Relation.Nullary.Negation using (contraposition)
+
+open MorphismStructures.ModuleMorphisms modA modB
+module A = Module modA
+open A using () renaming (Carrierᴹ to A)
+open Properties A.leftSemimodule
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴬ
+  ; non-zeroˡ to non-zeroˡᴬ
+  )
+module B = Module modB
+open B using () renaming (Carrierᴹ to B)
+open Properties B.leftSemimodule
+  using () renaming
+  ( non-zeroʳ to non-zeroʳᴮ
+  ; non-zeroˡ to non-zeroˡᴮ
+  )
+open CommutativeRing ring
+  using (_≈_; _*_; _+_) renaming
+  ( Carrier to S
+  ; 0#      to 𝟘
+  ; 1#      to 𝟙
+  )
+open Function.Definitions A._≈ᴹ_ B._≈ᴹ_
+
+_≉ᴬ_ : A → A → Set ℓm
+x ≉ᴬ y = ¬ (x A.≈ᴹ y)
+
+_≉ᴮ_ : B → B → Set ℓm
+x ≉ᴮ y = ¬ (x B.≈ᴹ y)
+
+module _
+  {f : A → B}
+  (isModuleHomomorphism : IsModuleHomomorphism f)
+  where
+
+  open IsModuleHomomorphism isModuleHomomorphism
+
+  -- f(x) ≈ 0 iff x ≈ 0, for linear non-trivial f
+  f𝟘≈𝟘 : f A.0ᴹ B.≈ᴹ B.0ᴹ
+  f𝟘≈𝟘 = begin
+    f A.0ᴹ          ≈⟨ ⟦⟧-cong (A.≈ᴹ-sym (A.*ₗ-zeroˡ A.0ᴹ)) ⟩
+    f (𝟘 A.*ₗ A.0ᴹ) ≈⟨ *ₗ-homo 𝟘 A.0ᴹ ⟩
+    𝟘 B.*ₗ f A.0ᴹ   ≈⟨ B.*ₗ-zeroˡ (f A.0ᴹ) ⟩
+    B.0ᴹ ∎
+    where open Reasoning B.≈ᴹ-setoid
+
+  x≈𝟘→fx≈𝟘 : {x : A} → x A.≈ᴹ A.0ᴹ → f x B.≈ᴹ B.0ᴹ
+  x≈𝟘→fx≈𝟘 {x = x} x≈𝟘 = begin
+    f x    ≈⟨ ⟦⟧-cong x≈𝟘 ⟩
+    f A.0ᴹ ≈⟨ f𝟘≈𝟘 ⟩
+    B.0ᴹ ∎
+    where open Reasoning B.≈ᴹ-setoid
+
+  fx≉𝟘→x≉𝟘 : {x : A} → f x ≉ᴮ B.0ᴹ → x ≉ᴬ A.0ᴹ
+  fx≉𝟘→x≉𝟘 = contraposition x≈𝟘→fx≈𝟘
+
+  -- Zero is a unique output of linear map ≉ `const 𝟘`.
+  zero-unique : {x : A} →
+    Σ[ (s , y) ∈ S × A ] ((s A.*ₗ x A.≈ᴹ y) × (f y ≉ᴮ B.0ᴹ)) →
+    x ≉ᴬ A.0ᴹ → f x ≉ᴮ B.0ᴹ
+  zero-unique {x = x} ((s , y) , (s·x≈y , fy≉𝟘)) x≉𝟘 =
+    non-zeroʳᴮ s·fx≉𝟘
+    where
+    open Reasoning B.≈ᴹ-setoid
+    y≉𝟘     : y ≉ᴬ A.0ᴹ
+    y≉𝟘     = fx≉𝟘→x≉𝟘 fy≉𝟘
+    fs·x≈fy : f (s A.*ₗ x) B.≈ᴹ f y
+    fs·x≈fy = ⟦⟧-cong s·x≈y
+    s·fx≈fy : s B.*ₗ f x B.≈ᴹ f y
+    s·fx≈fy = begin
+      s B.*ₗ f x   ≈⟨ B.≈ᴹ-sym (*ₗ-homo s x) ⟩
+      f (s A.*ₗ x) ≈⟨ fs·x≈fy ⟩
+      f y ∎
+    s·fx≉𝟘  : (s B.*ₗ f x) ≉ᴮ B.0ᴹ
+    s·fx≉𝟘  = λ s·fx≈𝟘 →
```

Comment:
No need to have a lambda here, this can go on the LHS of the equals, e.g. `s·fx≉𝟘 s·fx≈𝟘 =`

Reply:
Changed as requested.